test/support/test_repo.exs

```diff
@@ -108,6 +108,11 @@ defmodule Ecto.TestAdapter do
 
   ## Migrations
 
+  def lock_for_migrations(repo, opts, fun) do
+    send self(), {:lock_for_migrations, fun}
+    {:ok, fun.()}
```

Comment:
If we call `fun.()` with the migration versions, we can probably get rid of [this hack](https://github.com/blatyo/ecto/blob/4808d1381a61e42c64e4cb5b798fb1f0e271bc16/test/support/test_repo.exs#L38-L41) and move it down here - which is a bit more appropriate.